src/Microsoft.AspNet.OData.Shared/DeltaDeletedEntityObjectOfT.cs

```diff
@@ -0,0 +1,110 @@
+        /// </param>         
+        /// <param name="instanceAnnotationsPropertyInfo">The property info that is used as container for Instance Annotations</param>
+        public DeltaDeletedEntityObject(Type structuralType, PropertyInfo instanceAnnotationsPropertyInfo)
+            : this(structuralType, dynamicDictionaryPropertyInfo: null, instanceAnnotationsPropertyInfo)
+        {
+
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="DeltaDeletedEntityObject{TStructuralType}"/>.
+        /// </summary>
+        /// <param name="structuralType">The derived entity type or complex type for which the changes would be tracked.
+        /// <paramref name="structuralType"/> should be assignable to instances of <typeparamref name="TStructuralType"/>.
+        /// </param>
+        /// <param name="dynamicDictionaryPropertyInfo">The property info that is used as dictionary of dynamic
+        /// properties. <c>null</c> means this entity type is not open.</param>        
+        /// <param name="instanceAnnotationsPropertyInfo">The property info that is used as container for Instance Annotations</param>
+        public DeltaDeletedEntityObject(Type structuralType, PropertyInfo dynamicDictionaryPropertyInfo, PropertyInfo instanceAnnotationsPropertyInfo)
+            : this(structuralType, updatableProperties: null , dynamicDictionaryPropertyInfo, instanceAnnotationsPropertyInfo)
+        {
+ 
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="DeltaDeletedEntityObject{TStructuralType}"/>.
+        /// </summary>
+        /// <param name="structuralType">The derived entity type or complex type for which the changes would be tracked.
+        /// <paramref name="structuralType"/> should be assignable to instances of <typeparamref name="TStructuralType"/>.
+        /// </param>
+        /// <param name="updatableProperties"> Properties that can be updated</param>
+        /// <param name="dynamicDictionaryPropertyInfo">The property info that is used as dictionary of dynamic
+        /// properties. <c>null</c> means this entity type is not open.</param>
+        /// <param name="instanceAnnotationsPropertyInfo">The property info that is used as container for Instance Annotations</param>
+        public DeltaDeletedEntityObject(Type structuralType, IEnumerable<string> updatableProperties, PropertyInfo dynamicDictionaryPropertyInfo, PropertyInfo instanceAnnotationsPropertyInfo)
+            : base(structuralType, updatableProperties, dynamicDictionaryPropertyInfo, instanceAnnotationsPropertyInfo)
+        {
+            DeltaKind = EdmDeltaEntityKind.DeletedEntry;
+        }
+
+     
+        /// <inheritdoc />
+        public Uri Id { get; set; }
+        
+        /// <inheritdoc />
+        public DeltaDeletedEntryReason? Reason { get; set; }
+        
+        /// <inheritdoc />
+        public IEdmNavigationSource NavigationSource { get; set; }  
```

Comment:
>NavigationSource [](start = 36, length = 16)

Is this ever used?  In 4.0, a (added/changed/deleted) entity at the top level may reference a related entity set, but I don't think we're supporting that scenario in this release.  From the code, it looks like we check this property but I didn't see anywhere that it was ever set.

If we did support this property, I would think we would support it on DelteOfTStructuralType, but since we don't have any code that handles it then it's probably better to error upon deserialization if we encounter a Resource/DeletedResource from a different feed.

Reply:
edmDeltaDeletedEntity has  a NavigationSource which is used in serializer. So is it different there